tinymce-single-react-ui/src/external/wp-blocks/editable/index.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * External dependencies
+ */
+import { createElement, Component } from 'wp-elements';
+import tinymce from 'tinymce';
+import { isEqual, last } from 'lodash';
+
+function initialize( node, inline, onSetup ) {
+	if ( ! node ) {
+		return;
+	}
+
+	const config = {
+		target: node.querySelector( '[contenteditable=true]' ),
+		theme: false,
+		inline: true,
+		toolbar: false,
+		skin_url: '//s1.wp.com/wp-includes/js/tinymce/skins/lightgray',
+		entity_encoding: 'raw',
+		setup: onSetup,
+		formats: {
+			strikethrough: { inline: 'del' }
+		}
+	};
+
+	if ( inline ) {
+		config.valid_elements = '#p,br,b,i,strong,em,del,a[href|target]';
+	}
+
+	tinymce.init( config );
+}
+
+export default class EditableComponent extends Component {
+	static defaultProps = {
+		onChange: () => {},
+		splitValue: () => {},
+		onFocusChange: () => {},
+		onType: () => {},
+		initialContent: '',
+		inline: false,
+		single: false,
+	};
+
+	componentDidMount() {
+		initialize( this.node, this.props.inline, this.onSetup );
+		if ( this.props.focusConfig ) {
+			this.focus();
+		}
+	}
+
+	updateContent() {
+		// This could not be called on each content change, it used to change the cursor position
+		let bookmark;
+		if ( this.props.focusConfig ) {
+			bookmark = this.editor.selection.getBookmark( 2, true );
+		}
+		this.editor.setContent( this.props.content );
+		if ( this.props.focusConfig ) {
+			this.editor.selection.moveToBookmark( bookmark );
+		}
+	}
```

Comment:
I see that this code comes from the `per-block` prototype. My problem here is that I was not able to have a "controlled" TinyMCE, which means I was not able to call `setContent` each time the `content` prop changes without loosing the cursor position.

My second problem is that I was not able to get a reliable `onChange` event which is essential in a state based architecture. the TinyMCE's `onChange` is not triggered on each change and even when it was called, I think the content I got which `editor.getContent` was different from the real content at that moment.

Maybe you could have more expertise on these?